analysis_schema/base_model.py

```diff
@@ -168,9 +168,20 @@ def _run(self):
 
         # if there is a dataset sitting in _instantiated_datasets, add it to
         # the args and call as a keyword argument
-        if len(_instantiated_datasets) > 0:
-            ds_keys = list(_instantiated_datasets.keys())
-            ds = _instantiated_datasets[ds_keys[0]]
+
+        # if len(Dataset_Fixture.all_data) > 0:
+        #     ds_keys = list(Dataset_Fixture.all_data.keys())
+        #     for key in ds_keys:
+        #         ds = yt.load(Dataset_Fixture.all_data[key])
+        #         return val(*the_args, ds=ds)
+        # else:
+        #     raise AttributeError(
+        #         "could not find a dataset: cannot build the data container"
+        #     )
+
+        if len(dataset_fixture._instantiated_datasets) > 0:
```

Comment:
I think this is an example of where the code got a little confusing because we were mapping to the yt api directly -- instead of just pulling the first entry in the dataset_fixture, it might be a spot where it's actually clearer to have the pydantic model diverge a bit form the yt-api. Maybe by adding a dataset attribute to this class so that the `_run` function here could simply extract the specified dataset from the dataset_fixture ?

Reply:
don't need to worry about this one for this PR!